File: frontend/scripts/sync-data.mjs
// Copies the pipeline's real output (../data/processed/opportunity_areas.json)
// into frontend/data/, which is what the deployed app actually reads (see
// lib/data.ts). Run this after every pipeline/synthesize.py run you want to
// publish — it's a manual/explicit step, not automatic, so a deploy never
// silently picks up a run you didn't intend to publish.
import { copyFile, mkdir } from "fs/promises";
import path from "path";
import { fileURLToPath } from "url";

const frontendDir = path.dirname(path.dirname(fileURLToPath(import.meta.url)));
const src = path.join(frontendDir, "..", "data", "processed", "opportunity_areas.json");
const destDir = path.join(frontendDir, "data");
const dest = path.join(destDir, "opportunity_areas.json");

await mkdir(destDir, { recursive: true });
await copyFile(src, dest);
console.log(`[sync-data] copied ${src} -> ${dest}`);

File: backend/sync_data.py
"""Copies the pipeline's real output (../data/processed/opportunity_areas.json)
into backend/data/, which is what the deployed API actually serves (see
main.py). Run this after every pipeline/synthesize.py run you want to
publish — a manual, explicit step so a deploy never silently picks up a run
you didn't intend to publish.
"""
import shutil
from pathlib import Path

SRC = Path(__file__).resolve().parent.parent / "data" / "processed" / "opportunity_areas.json"
DEST_DIR = Path(__file__).resolve().parent / "data"
DEST = DEST_DIR / "opportunity_areas.json"

if __name__ == "__main__":
    if not SRC.exists():
        raise SystemExit(f"[sync_data] ERROR: {SRC} not found — run the pipeline first.")
    DEST_DIR.mkdir(parents=True, exist_ok=True)
    shutil.copy2(SRC, DEST)
    print(f"[sync_data] copied {SRC} -> {DEST}")
